### DISCUSSION TITLE
Opening v3.98.9 built from sources on macos m1 crashes with "Cannot destructure property 'promisify' of 'fh' as it is undefined"

### DISCUSSION THREAD

**Discussion by hkjn** [tutao/tutanota#4424] [General]
- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

**Describe the bug**
On macos 12.4, macbook m1 pro, running 3.98.9 built from source crashes on startup.

**To Reproduce**
1. Try to build `tutanota-release-3.98.9` from source
2. Install `.dmg`
3. Attempt to launch app

**Expected behavior**
Desktop app launches without crash.

**Desktop (please complete the following information):**
 - OS: macOS v12.4
 - npm v8.12.1
 - node v18.5.0

**Additional context**

I first applied patch @charlag suggested from https://github.com/tutao/tutanota/issues/4091 to update architecture to `arm64`: https://github.com/hkjn/tutanota/commit/a099a5ade01b2b8ce966bc425bc3ce4c97461663.

Error when launching app is:

```
Uncaught Exception:
TypeError: Cannot destructure property 'promisify' of 'fh' as it is undefined.
at Object.<anonymous> (/Applications/Tutanota 
Desktop.app/Contents/Resources/app.asar/desktop/DesktopMain.js:1:829030)
at Module._compile (node:internal/modules/cjs/loader:1118:14)
at Module._extensions..js (node:internal/modules/cjs/loader:1173:10)
at Module.load (node:internal/modules/cjs/loader:988:32)
at Module._load (node:internal/modules/cjs/loader:829:12)
at c._load (node:electron/js2c/asar_bundle:5:13343)
at Object.<anonymous> (node:electron/js2c/browser_init:185:3104)
at ./lib/browser/init.ts (node:electron/js2c/browser_init:185:3308)
at __webpack_require__ (node:electron/js2c/browser_init:1:128)
at node:electron/js2c/browser_init:1:1200
```

**Comment by charlag** [comment]
Hi
It is not enough to just swap target architecture, I answered this for https://github.com/tutao/tutanota/issues/4413 already.

**Reply by hkjn** [reply]
Thanks for the reply, @charlag.

Pulling together the info from your replies on #4413, the hints for how to produce the desktop app from source code for MacOS with ARM64 is:

```
It is not enough to just swap target architecture
If you want to build with ARM as a target you would need to give it correct OpenSSL build in some way.
You would need to fork our fork of better-sqlite3-sqlcipher to add arm prebuild instead of x86 prebuilt of openssl there and then it should just work
```

So can we get https://github.com/tutao/tutanota/blob/master/doc/BUILDING.md updated to describe these steps in enough detail so that they can be followed?

How were the releases on https://tutanota.com for MacOS produced?

**Reply by charlag** [reply]
Please see #4439 for hints on ARM builds.

Once we will be supporting ARM first-party no additional steps will be necessary.
